crates/bitcoind_rpc/src/bip158.rs

```diff
@@ -177,16 +177,18 @@ impl<'c, C: RpcApi> Iterator for FilterIter<'c, C> {
     type Item = Result<Event, Error>;
 
     fn next(&mut self) -> Option<Self::Item> {
+        if self.spks.is_empty() {
+            return None;
+        }
```

Comment:
I assume that the user would properly use the `extend` or `add` methods, however, I wonder how could we make it clear here that it's missing the `spks`, otherwise the user would only get Nones without much information 🤔

Reply:
In this case would it be better to return `Event::NoMatch`? If I recall, the reason for adding this check was that `FilterIter` seemed to be returning empty blocks as matches even without spks (on regtest) which is also confusing. I'll do some more testing though to confirm it.

Reply:
I would say it could be a specific error variant indicating the spks are empty.